cpp/src/arrow/compute/kernels/aggregate_test.cc

```diff
@@ -1625,6 +1669,10 @@ TEST_F(TestRandomQuantileKernel, Histogram) {
   // exercise histogram approach: size >= 65536, range <= 65536
   this->CheckQuantiles(/*array_size=*/80000, /*num_quantiles=*/100);
 }
+
+TEST_F(TestRandomQuantileKernel, TDigest) {
+  this->CheckTDigests(/*chunk_sizes=*/{12345, 6789, 8765, 4321}, /*num_quantiles=*/100);
```

Comment:
Does it also check the case whether there are only nulls or NaNs? If not, can you add explicit tests?

Reply:
Nulls are check. NaNs not.
Have changed test array to add both Nulls and NaNs.

Reply:
Hmm..., only Nulls or NaNs are not checked for TDigest, will add.

Reply:
Test added.